### PR TITLE
fix: if app has no subscriptions, then nothing to process

### PR DESCRIPTION
**Pull request by mayzhang2000**

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -339,6 +339,10 @@ func (c *notificationController) processQueueItem() (processNext bool) {
 	appCopy := app.DeepCopy()
 	logEntry := log.WithField("app", key)
 	logEntry.Info("Start processing")
+	subscriptions := c.getSubscriptions(appCopy)
```
Comment (alexmt) [inline controller/controller.go:342]: It is still possible to reproduce the same bug if the application has a subscription but non of the subscriptions triggers a notification. 

The bug is here: https://github.com/argoproj-labs/argocd-notifications/blob/8b9ee894bd976115d30d225d1ba29f378478ef1f/controller/controller.go#L351

If `app.GetAnnotations()` is nil then `appCopy.GetAnnotations()` is empty map. Empty annotations map and nil map should be considered the same.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: controller/controller.go (hunk 1 above)
Comment (alexmt) [reply controller/controller.go:342]: this is no longer required once annotation comparison is fixed

File: controller/controller.go [hunk 2]
```diff
@@ -348,7 +352,9 @@ func (c *notificationController) processQueueItem() (processNext bool) {
 		logEntry.Errorf("Failed to process: %v", err)
 		return
 	}
-	if !reflect.DeepEqual(app.GetAnnotations(), appCopy.GetAnnotations()) {
+
+	if !(app.GetAnnotations() == nil && len(appCopy.GetAnnotations()) == 0) &&
```
Comment (alexmt) [inline controller/controller.go:356]: Look like this catches only one case: before annotations map is nil and empty after. Due to refactoring we might end up with a situation where first map is empty and second is nil. Please create a method that compares two maps and reliably compare maps with both cases

**Review by alexmt — CHANGES_REQUESTED** [review]

File: controller/controller.go (hunk 2 above)
Comment (mayzhang2000) [reply controller/controller.go]: Good suggestion. Fix is checked in.

**Review by alexmt — APPROVED** [review]
LGTM. thank you!

**Pull request merged**